### PR TITLE
Fix subscription confirmation email when "Subscribe all customers to the newsletter" is enabled  978

### LINKED ISSUE #978: Fix subscription confirmation email when "Subscribe all customers to the newsletter" is enabled
### Preconditions

1. Magento 1 any version
2. Mailchimp for Magento 1.1.16

### Steps to reproduce
1. Enable double opt-in in Mailchimp and Magento
2. Enable option "Subscribe all customers to the newsletter" in Mailchimp for Magento
3.1 Place a new order using guest checkout method.
3.2 Alternatively abandon checkout after finishing second step.
### Expected result
1. Subscription confirmation email sent.

### Actual result
1. Subscription confirmation email is not sent.

### PR DESCRIPTION
**Pull request by brian-ebizmarts**
Fix subscription confirmation email when "Subscribe all customers to the newsletter" is enabled closes #978

### CODE REVIEW

File: app/code/community/Ebizmarts/MailChimp/Model/Api/Customers.php [hunk 1]
```diff
@@ -115,10 +115,11 @@ public function createBatchJson($mailchimpStoreId, $magentoStoreId)
                 $this->logCouldNotEncodeCustomerError($customer);
             }
 
-            $isSubscribed = $subscriber->loadByEmail($customer->getEmail())->getSubscriberId();
-
-            if ($this->optInStatusForStore && !$isSubscribed){
-                $subscriber->subscribe($customer->getEmail());
+            if($this->optInStatusForStore) {
```
Comment (Santiagoebizmarts) [inline app/code/community/Ebizmarts/MailChimp/Model/Api/Customers.php]: Add a blank space after **if** and before **(** for better readability

File: app/code/community/Ebizmarts/MailChimp/Model/Api/Orders.php [hunk 2]
```diff
@@ -833,4 +846,11 @@ protected function getApiProduct()
         return Mage::getModel('mailchimp/api_products');
     }
 
+    /**
+     * @return false|Mage_Core_Model_Abstract
```
Comment (Santiagoebizmarts) [inline app/code/community/Ebizmarts/MailChimp/Model/Api/Orders.php]: Change return type to Mage_Newsletter_Model_Subscriber

**Review by SantiParagarino — CHANGES_REQUESTED** [review]

File: app/code/community/Ebizmarts/MailChimp/Model/Api/Products.php [hunk 3]
```diff
@@ -783,10 +783,11 @@ public function getParentImageUrl($childId, $magentoStoreId)
      */
     protected function getProductUrl($product, $magentoStoreId)
     {
-        $oldStoreId = Mage::app()->getStore()->getId();
-        Mage::app()->setCurrentStore($magentoStoreId);
+        $helper = $this->getMailChimpHelper();
```
Comment (Santiagoebizmarts) [inline app/code/community/Ebizmarts/MailChimp/Model/Api/Products.php]: Is it necessary to do this or is it already done in the createBatchJson function?

**Review by centerax — CHANGES_REQUESTED** [review]
Please add a meaningful name to the PR.

**Review by SantiParagarino — APPROVED** [review]

**Review by SantiParagarino — APPROVED** [review]

**Review by centerax — APPROVED** [review]

**Pull request merged**